ci/github/helpers/openapi-specs-backwards-compatibility.bash

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Recursively checks if all openapi specs within a local osparc-simcore revision are backwards compatible with a remote base
+# Example: bash osparc-simcore/ci/github/helpers/openapi-specs-backwards-compatibility.bash \
+#    https://raw.githubusercontent.com/ITISFoundation/osparc-simcore/refs/heads/master
+#    ./osparc-simcore/
+
+base_remote=$1
+revision_local=$2
+
+repo_base_dir=$(realpath "$(dirname "${BASH_SOURCE[0]}")/../../..")
+openapi_specs=$(find "${revision_local}" -type f \( -name 'openapi.json' -o -name 'openapi.yaml' \) -not -path '*/.*' -exec realpath --relative-to="${revision_local}" {} \;)
+
+cd "${revision_local}" || exit 1 # required to mount correct dir for diff tool
+
+exit_status=0
+for spec in ${openapi_specs}; do
+  echo "Comparing ${spec}"
+  "${repo_base_dir}/scripts/openapi-diff.bash" breaking --fail-on ERR "${base_remote}/${spec}" "/specs/${spec}"
+  exit_status=$(("${exit_status}" + $?))
+  printf "%0.s=" {1..100} && printf "\n"
+done
+
```

Comment:
Yes, in that case I guess the CI will report errors and it will have to be forced in. I was a bit in doubt if it really makes sense to test backwards compatibility of all openapi-specs, since we control all the clients of the APIs which "internal" to the deploymnent. So we just need to fix those clients and then breaking backwards compatibility is not actually an issue. So maybe I change this test to only include the api-server(?) What do you think @sanderegg? Or maybe the api-server and web-server?

Reply:
As you can see [here](https://github.com/ITISFoundation/osparc-simcore/actions/runs/12141554291/job/33854199629?pr=6866#logs) this test might produce quite a signicficant amount of errors 😅

Reply:
well I guess api-server at least yes.
the others should be allowed but I still see the point of finding them out. not sure if it is possible to issue a warning instead of an error for these?

Reply:
Good point. I will investigate how to make the backwards compatibility errors appear as warnings in the pr instead of failing tests